integrations/firehydrant/client.py

```diff
@@ -0,0 +1,451 @@
+        """
+        logger.info(f"Getting tasks details for incident: {incident_id}")
+        params: dict[str, Any] = {
+            "page": 1,
+            "per_page": PAGE_SIZE,
+        }
+        task_url = f"incidents/{incident_id}/tasks"
+
+        tasks = await self.send_paginated_api_request(
+            endpoint=task_url, query_params=params, data_key="data"
+        )
+        task_completion_status = await self.get_task_completion_status(tasks)
+        return task_completion_status
+
+    async def get_task_completion_status(
+        self, tasks: list[dict[str, Any]]
+    ) -> dict[str, Any]:
+        """
+        Return statistics about completed and uncompleted tasks. It returns a dictionary with counts of completed and uncompleted tasks, providing insights into task progress.
+
+        Args:
+            tasks (list[dict[str, Any]]): A list of task objects, each represented as a dictionary.
+
+        Returns:
+            dict[str, Any]: A dictionary containing statistics about completed and uncompleted tasks.
+        """
+        logger.info("Computing task completion statistics")
+        total_completed_tasks = 0
+        for task in tasks:
+            if task.get("state") == "done":
+                total_completed_tasks += 1
+
+        return {
+            "completed": total_completed_tasks,
+            "incompleted": len(tasks) - total_completed_tasks,
+        }
+
+    async def compute_incident_duration(
+        self, milestones: list[dict[str, Any]]
+    ) -> Optional[float]:
+        """Compute the duration in minutes it took to resolve an incident based on milestones.
+
+        Args:
+            milestones (list[dict[str, Any]]): A list of milestone objects representing incident events.
+
+        Returns:
+            float: The duration in minutes it took to resolve the incident.
+        """
+        started_time = None
+        resolved_time = None
+
+        # Iterate through the milestones to find the "started" and "resolved" events.
+        for milestone in milestones:
+            event_type = milestone.get("type")
+            occurred_at = milestone.get("occurred_at", "")
+
+            # Check if the event is a "started" event.
+            if event_type == "started":
+                started_time = datetime.fromisoformat(occurred_at[:-1])
+
+            # Check if the event is a "resolved" event.
+            if event_type == "resolved":
+                resolved_time = datetime.fromisoformat(occurred_at[:-1])
+
+        # Calculate the duration if both "started" and "resolved" events are found.
+        if started_time and resolved_time:
+            duration = resolved_time - started_time
+            duration_hours = duration.total_seconds() / 60  # Convert to minutes
+            return round(duration_hours, 2)
+
+        return None  # Return None if either "started" or "resolved" event is missing.
```

Comment:
i dont think there is a need for this calculation

it can be done by sending the milestones and calculating the data in calculation property

Reply:
I'm not really familiar with the calculation property that's why I did it in code. Can you point me to a link where this can be done?